src/classes/HouseholdNamingService.cls

```diff
@@ -0,0 +1,512 @@
+/*
+    Copyright (c) 2011, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2011 (2.3)
+* @group Households
+* @description Controls generation of household names for lists of households
+* NOTE: This class handles both household naming for the Household__c object
+* as well as for HH Account objects.
+*/
+public without sharing class HouseholdNamingService {
```

Comment:
At some point we should understand whether the naming classes truly require "without". I would expect them to be called by either a) a trigger; or b) a batch operation. In both cases, "inherited" sharing would be sufficient. I'm a bit nervous about changing the sharing declaration on an existing class, but as with the above comment I'm not clear on whether this is new or not.

Reply:
This class is renamed from HH_HouseholdNaming - the sharing declaration is unchanged.  There is a public method in this class that executes the batch to refresh all household naming.  Whatever in here needs to truly be without should probably be factored out into its own smaller class.  But would need to do some more investigation to figure out what really needs to have without sharing.

Reply:
To add to this comment, this class is going to need some further refactoring for sure. It is in an okay place for now but besides the "without sharing" logic being factored out into its own class, this class queries household members and have some other stuff that should probably be pushed into the domain classes instead of here in the service class. I'm going to leave it for now in the interest of stabilizing this branch to finalize review and testing.